File: dotnet/Kata/Model/Currency.cs
using System.ComponentModel;
using Kata.External;

namespace Kata.Model;

public enum Currency
{
    [Description("€")]
    Euro,
    [Description("$")]
    Dollar,
    [Description("£")]
    Pound,
    [Description("¥")]
    Yen
}

public static class CurrencyExtensionMethods {

    public static CurrencyIsoCode ToIsoCode(this Currency currency)
    {
        return currency switch
        {
            Currency.Euro => CurrencyIsoCode.EUR,
            Currency.Dollar => CurrencyIsoCode.USD,
            Currency.Pound => CurrencyIsoCode.GBP,
            Currency.Yen => CurrencyIsoCode.JPY,
            _ => throw new ArgumentOutOfRangeException(nameof(currency), currency, null)
        };
    }
}

File: python/src/currency.py
from enum import Enum

from src.external.currency_iso_code import CurrencyIsoCode


class Currency(Enum):
    Euro = 1
    Dollar = 2
    Pound = 3
    Yen = 4

    def to_iso_code(self):
        if self == Currency.Euro:
            return CurrencyIsoCode.EUR
        elif self == Currency.Dollar:
            return CurrencyIsoCode.USD
        elif self == Currency.Pound:
            return CurrencyIsoCode.GBP
        elif self == Currency.Yen:
            return CurrencyIsoCode.JPY
        pass
